course_discovery/populate_course_run_reporting_type_and_announcement.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import datetime
+from dateutil.parser import parse
+import json
+import logging
+import os
+
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+
+logger = logging.getLogger()
+
+
+class CatalogApiService(object):
+    """The service to interface with edX catalog API"""
+
+    def __init__(self, oauth_host, oauth_key, oauth_secret, api_url_root):
+        try:
+            self.access_token, expires = EdxRestApiClient.get_oauth_access_token(
+                '{root}/access_token'.format(root=oauth_host),
+                oauth_key,
+                oauth_secret, token_type='jwt'
+            )
+        except Exception:
+            logger.exception('No access token acquired through client_credential flow.')
+            raise
+
+        logger.info('Token retrieved: %s', access_token)
+
+        self.api_client = EdxRestApiClient(api_url_root, jwt=self.access_token)
+
+    def _get_resource_from_api(self, api_endpoint, page_size=None, **kwargs):
+        page = 1
+        results = []
+
+        while page >= 1:
+            if page_size:
+                response = api_endpoint.get(limit=page_size, offset=((page - 1) * page_size), **kwargs)
+            else:
+                response = api_endpoint.get(page=page, **kwargs)
+            if response.get('next'):
+                page += 1
+            else:
+                page = -1
+            results.extend(response.get('results'))
+
+        return results
+
+def process_line(line):
+    key, reporting_type, announcement = line.strip().split(',')
+    data = {
+        'reporting_type': reporting_type
+    }
+    if announcement != '':
+        # Convert 2/1/2016 to 2016-02-01T00:00:00+00:00
```

Comment:
Personally, I wasn't exactly sure what formatting these did until I wrote this, to me this would be helpful if I don't already know what isoformat is which i didn't